add blogpost on migrating away from the Jaeger exporter (#3273)
open-telemetry/opentelemetry.io: content/en/blog/2023/jaeger-exporter-collector-migration.md

---
title: Migrating away from the Jaeger exporter in the Collector

> linkTitle: Migrating away from the Jaeger exporter in the Collector

Reviewer: ```suggestion
linkTitle: Jaeger Collector Exporter Migration
```

The `linkTitle` needs to be fairly short, not 100% happy with my suggestion, but it's the first that came to my mind

> date: 2023-09-14

Reviewer: Let's aim for some time next week, we still have another blog post in the loop for Monday next week, maybe Wednesday/Thursday?

Author: That's fine, the release was done Monday and there was an issue raised yesterday about it, which is why I figured the sooner the better :)

author: >-

>   [Alex Boten](https://twitter.com/codeboten) (Lightstep)

Author: ```suggestion
  [Alex Boten](https://github.com/codeboten) (Lightstep)
```

>   [Alex Boten](https://twitter.com/codeboten) (Lightstep)

Reviewer: I think so far we always linked github accounts.

Author: I just followed the example i copied here: https://github.com/open-telemetry/opentelemetry.io/blob/1fb0f484fbac6df101139f7db789fbb306eb0f6e/content/en/blog/2023/outreachy-may-cohort.md?plain=1#L6

I don't have a problem changing it though.

Reviewer: Ah... I missed that one. Whatever you prefer, I personally prefer the github accounts because (a) they give people direct insights into your work with OpenTelemetry and (b) your twitter link only works for people authenticated with the platform

Author: np. I like the idea of being consistent :)

cSpell:ignore:
---

The
[latest binary](https://github.com/open-telemetry/opentelemetry-collector-releases/releases/tag/v0.85.0)
release of the OpenTelemetry Collector no longer includes exporters for the
native Jaeger format. This change was prompted because Jaeger has support for
OTLP out of the box. If you are a current user of either the Jaeger or Jaeger
Thrift exporter, you have a couple of options to choose from when upgrading to
the latest Collector.

## Switch to OTLP exporter

The recommended option is to switch to the OpenTelemetry Protocol exporter. This
exporter is included in all the official distributions supported by the
community. To switch, you'll need to update the collector configuration. Find
the following Jaeger exporter configuration:

```yaml
exporters:
  jaeger:
    endpoint: https://jaeger.example.com:14250

service:
  pipelines:
    exporters: [jaeger]
```

And replace it with the OTLP configuration:

```yaml
exporters:

>   otlp/jaeger: # Jaeger supports OTLP directly

Reviewer 2: Maybe worth explaining the port (and mentioning in passing this is OTLP/gRPC specifically)
```suggestion
  otlp/jaeger: # Jaeger supports OTLP directly. The default port for OTLP/gRPC is 4317
```

>     endpoint: https://jaeger.example.com:4317

Reviewer 3: TLS is not enabled in Jaeger by default, it's worth calling out here.

service:
  pipelines:
    exporters: [otlp/jaeger]
```

Depending on your version of Jaeger, you may need to set the following

> environment variable `COLLECTOR_OTLP_ENABLED=true`.

Reviewer: can you put a bound here like "since version X this is not required anymore"

Author: This is a good question, maybe @pavolloffay or @yurishkuro can confirm what the right versions are here. I onlny tested this w/ the all-in-one container versions 1.35 & 1.41.

Reviewer 3: from v1.46.0 there is no need to set COLLECTOR_OTLP_ENABLED=true

> Depending on your version of Jaeger, you may need to set the following
> environment variable `COLLECTOR_OTLP_ENABLED=true`.

Author: ```suggestion
Versions of Jaeger until v1.46.0, need the following environment
variable set `COLLECTOR_OTLP_ENABLED=true`.
```

## Build a custom Collector

If switching to the OTLP exporter isn't an option, an alternative is to build a
custom Collector that includes the Jaeger exporter. The process to build is
documented [here](https://opentelemetry.io/docs/collector/custom-collector/).
Your manifest file will need to include the following line to add the Jaeger
exporter:

```yaml
exporters:
  - gomod: go.opentelemetry.io/collector/exporter/jaegerexporter v0.85.0
  - gomod:
      go.opentelemetry.io/collector/exporter/jaegerthrifthttpexporter v0.85.0
```

Further example of manifest files for existing Collector distributions can be
found in the
[opentelemetry-collector-releases](https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol)
repository.

If either of these options do not work for your use-case, please reach out to

> the #otel-collector community via [CNCF Slack](https://slack.cncf.io) or open an

Reviewer: ```suggestion
the [#otel-collector](https://cloud-native.slack.com/archives/C01N6P7KR6W) community via [CNCF Slack](https://slack.cncf.io) or open an
```

issue in the

> [repository](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues).

Reviewer 2: ```suggestion
[repository](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/new/choose).
```
